src/hooks/useSelectRow.js

```diff
@@ -0,0 +1,56 @@
+import { useMemo, useState, useEffect, useCallback } from 'react'
+
+export const useSelectRow = props => {
+  const { rows, data } = props
+  const [selected, setSelected] = useState([])
```

Comment:
The state living here technically works, but similar to other hooks, I think it should use the `useTableState` initial state and be hoistable like all of the other state.

Reply:
Take a look at the usePagination hook to see how that's done.